### DISCUSSION TITLE
Why it is too difficult to install podman !!!

### DISCUSSION THREAD

**Discussion by distributed-clouds** [containers/podman#23182] [General]
I've been trying to switch from Docker to Podman, but the installation process for Podman is surprisingly challenging compared to Docker. While Docker provides easy-to-use compiled binary versions (e.g., https://download.docker.com/linux/static/stable/aarch64/), Podman's GitHub releases page only offers podman-remote. For instance, on Amazon Linux 2023, installing Podman requires building multiple binaries like conmon and other dependencies. This complexity is a significant barrier for adoption. Why doesn't the Podman team provide static binaries in their releases, similar to Docker's approach? A simpler installation process would greatly encourage more users to try Podman as an alternative to Docker. The current situation is frustrating for those of us who want to work with Podman but find the setup process unnecessarily complicated.

**Comment by giuseppe** [comment]
installing static binaries seems convenient initially but then you'll have to deal with other issues like software updates.  How are you going to make sure your packages are updated and not end up with a software vulnerability because you forgot to check if there is a new binary available?

Packaging and providing Podman binaries is the distro responsibility.  I am not familiar with Amazon Linux, but since it is based on Fedora, it should not be too hard to provide the Podman rpms.

**Reply by afbjorklund** [reply]
Amazon Linux is a fork of Fedora Linux, where they have replaced Podman with Docker:

https://docs.aws.amazon.com/linux/al2023/ug/relationship-to-fedora.html

I don't think they will add it back, but you _could_ probably recompile the source packages:

* https://github.com/amazonlinux/amazon-linux-2023/issues/67

**Reply by distributed-clouds** [reply]
> Amazon Linux is a fork of Fedora Linux, where they have replaced Podman with Docker:
> 
> https://docs.aws.amazon.com/linux/al2023/ug/relationship-to-fedora.html
> 
> I don't think they will add it back, but you _could_ probably recompile the source packages:
> 
> * [[Package Request] - Buildah/Podman/Skopeo amazonlinux/amazon-linux-2023#67](https://github.com/amazonlinux/amazon-linux-2023/issues/67)

Hi @afbjorklund, 
yes I have recompiled it on my ec2, but again the  problem is not about the amazon linux or any other distro!!! 
what Im saying is that the binaries should be available as easy as the docker binaries. this helps a lot of people to adopt podman in their pipeline

**Reply by afbjorklund** [reply]
The static builds were too much work to maintain, so they were [removed](https://github.com/containers/podman/commit/346beec35a11393e9c59278d47b18de337c2b06f) in Podman v4.1

Same goes with [the packages for Ubuntu](https://packages.ubuntu.com/podman), that is all up to Debian now... (no more OBS). It's there in the system, but an older version.

**Reply by rhatdan** [reply]
Podman is an open source project not a product.  As an open source project we rely on distros to package the tools for specific distros. Podman relies on many other packages within the Distro crun, netavark, pasta and many others. Packaging these up for every possible distro and every possible version is not something the upstream project is willing to do.  Some companies, Red Hat and OpenSUSE for example pay people to maintain versions for their distros.  If Amazon Linux (Often the largest company in the world) chooses to not do this for you, then your complaint should be with them, not the upstream project.

**Reply by rhatdan** [reply]
Of course you could also switch to a distribution that keeps Podman up 2 date like Fedora.

**Reply by MushiTheMoshi** [reply]
then at least provide the full instructions to build it properly per version, I have tried following(https://podman.io/docs/installation#building-from-source)  using debian and ubuntu without success, so many missing dependencies and things that not work.

**Reply by afbjorklund** [reply]
Instruction for distributions are available at: https://github.com/containers/podman/blob/main/DISTRO_PACKAGE.md

The instructions on the website are not being updated anymore, see https://github.com/containers/podman.io/issues

**Comment by distributed-clouds** [comment]
Hi @giuseppe Thanks for you message.
Thats exactly what I said, I don't think this state is good for the end user: 
**Packaging and providing Podman binaries is the distro responsibility**

In my opinion user should be able to grab podman binaries as easy as possible.

**Comment by erlangparasu** [comment]
try: https://github.com/mgoltzsche/podman-static
